Update Blacklist-Whitelist Recommendation (#114)
inclusivenaming/website: content/word-lists/tier-1.md

## Whitelist, blacklist

**Definition:** 
### Terms

Lists which permit or deny a set of nouns, or select enabled features.
* blacklist
* whitelist

**Proposed alternative:** 
### Concept

* allowlist/denylist 
* allowedNouns/deniedNouns
Noun: A list of authorized or unauthorized entities that might be permitted or denied access to computers and networks.

**Reasoning:** 
Verb: The computer administration practice of creating a list of authorized or unauthorized entities to protect computers and networks and to prevent unnecessary demand for resources.

The underlying assumption of the whitelist/blacklist metaphor is that white = good and black = bad. 
Because colors in and of themselves have no predetermined meaning, any meaning we assign to them is cultural: for example, the color red in many Southeast Asian countries is lucky, and is often associated with events like marriages, whereas the color white carries the same connotations in many European countries. 

> In the case of whitelist/blacklist, the terms originate in the publishing industry – one dominated by the USA and England, two countries which participated in slavery and which grapple with their racist legacies to this day.

Reviewer: If countries will be named, use "the United Kingdom" instead of "England".

* allowlist/allow

**Admitted replacement terms**

* passlist
* deny

### Unsuitable replacement terms

### Use context

* blocklist or denylist: A list of entities that are denied access
* allowlist or passlist: A list of entities that are permitted access

### Rationale for replacement

These terms are based on a metaphor where black represents bad and white represents good, which reinforces systemic racial bias. Terms that use color representationally, such as “black box” and “white glove,” do not reinforce this discriminatory metaphor and are fine to use.

> From a technical communication perspective, using whitelist/blacklist as a naming convention applies metaphor (and, in turn, unintended meaning) when it isn’t needed. More directly descriptive words like blocklist/allowlist enhances understanding.

Reviewer: Change "enhances" > "enhance", for subject-verb agreement.
Alternatively: "The use of more directly descriptive words ... enhances understanding."

Reviewer 2: ```suggestion
From a technical communication perspective, using whitelist/blacklist as a naming convention applies metaphor (and, in turn, unintended meaning) when it isn’t needed. More directly descriptive words like blocklist/allowlist enhance understanding.
```

Also acceptable:

* Leader/follower
* Parent/child

**Reasoning:**

As the IETF puts it, “Master-slave is an oppressive metaphor that will and should never become fully detached from history.”
The word’s origins and historicial use reveal use that is at best chauvinistic and racist, and in almost all cases connotative of ownership.
While there is some small ambiguity about the term master, the term slave is unambiguously about the ownership and subjugation of another person, and has been since its inception.

> The terms master/slave are harmful to Black and people of color contributors and employees.

Reviewer 2: I'm not sure using "employee" in a general world context makes as much sense, additionally people who are paid to contribute to Open Source software are still contributors.

```suggestion
The terms master/slave are harmful to Black and people of color contributors.
```

Reviewer 3: Those are all good changes, Karsten. Thanks.

Reviewer 3: That's a good suggestion, Mark. Your revised wording is much better.

Reviewer 4: @quaid sounds good. I think `indigenous` should be capatilized...proposed fix below.
```suggestion
The master/slave terminology normalizes and perpetuates the historical practice of enslavement.
Additionally, the terms master/slave are directly harmful to Black, Indigenous, and people of color (BIPOC).
```

* [IETF Network Working Group: Terminology, Power and Oppressive Language](https://tools.ietf.org/id/draft-knodel-terminology-00.html)
* [django/django PR #2692](https://github.com/django/django/pull/2692)
* [Python issue #34605](https://bugs.python.org/issue34605)
* [Drupal issue](https://www.drupal.org/node/2275877)
* [COUCHDB-2248](https://issues.apache.org/jira/browse/COUCHDB-2248)

## Master

> In computing, A “master” often refers to the original or main version of an entity.

Reviewer: Use lowercase "a".

* main
* original
* source
* control plane

**Reasoning:**

> While master in and of itself is potentially neutral, the propensity in which it is associated with the term slave in computing makes master on its own guilty by association. Though it is used as a standalone, it's impossible to remove the association with command and control entirely, and thus we recommend moving away from even singular use.

Reviewer: Use "stand-alone" as an adjective.

> While master in and of itself is potentially neutral, the propensity in which it is associated with the term slave in computing makes master on its own guilty by association. Though it is used as a standalone, it's impossible to remove the association with command and control entirely, and thus we recommend moving away from even singular use.

Reviewer: What about "master" in the context of mastery of a skill?
